src/FranceCities/FranceCities.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace App\FranceCities;
+
+use App\Entity\Geo\Zone;
+
+class FranceCities
+{
+    private static $cities = [];
+    private static $citiesByInseeCode = [];
+    private CitiesStorageInterface $citiesStorage;
+
+    public static $countries = [
+        '98000' => 'MC', // Monaco
+        '971' => 'GP', // Guadeloupe
+        '972' => 'MQ', // Martinique
+        '973' => 'GF', // Guyane
+        '974' => 'RE', // Réunion
+        '975' => 'PM', // Saint-Pierre-et-Miquelon
+        '976' => 'YT', // Mayotte
+        '986' => 'WF', // Wallis-et-Futuna
+        '987' => 'PF', // Polynésie
+        '988' => 'NC', // Nouvelle Calédonie
+    ];
+
+    public function __construct(CitiesStorageInterface $citiesStorage)
+    {
+        $this->citiesStorage = $citiesStorage;
+    }
+
+    public function findCitiesByPostalCode(string $postalCode): array
+    {
+        return $this->findCitiesForPostalCode($this->getCitiesList(), $postalCode);
+    }
+
+    public function getCitiesByInseeCode(): array
+    {
+        if (empty(self::$citiesByInseeCode)) {
+            foreach ($this->getCitiesList() as $inseeCode => $city) {
+                self::$citiesByInseeCode[str_pad($inseeCode, 5, '0', \STR_PAD_LEFT)] = $city['name'];
+            }
+        }
+
+        return self::$citiesByInseeCode;
+    }
```

Comment:
what is the purpose of this method as the initial `self::$cities` is already indexed by `insee_code`?